src/c2goto/library/io.c

```diff
@@ -37,6 +37,30 @@ __ESBMC_HIDE:;
   return ret;
 }
 
+// Reads characters from the standard input (stdin)
+// and stores them as a C string into str until a newline character
+// or the end-of-file is reached.
+// Source: https://www.cplusplus.com/reference/cstdio/gets/
+char *gets(char *str)
+{
+__ESBMC_HIDE:;
+  size_t i = 0;
+  size_t size = nondet_uint();
+  for(i = 0; i < size; i++)
+  {
+    char character = nondet_char();
+    if(character == '\n' || character == EOF)
```

Comment:
Remove `|| character == EOF`: this would fail when `(unsigned char)character == 0xff`. As `size` is already non-det. here, it should be fine not to check for EOF (until we properly model that).

Reply:
Fixed.